### ISSUE TITLE
Test detection fails when using `typeof someFunction<T>`

### ISSUE THREAD

**Issue by cmd-johnson** [vitest-dev/vscode#182]
**Describe the bug**
When using the `typeof` operator to get the type of a generic function including its type parameters anywhere in a test file, no further tests are detected in that file. The console log just reads "parse error".

**To Reproduce**
Steps to reproduce the behavior on [the example project](https://github.com/zxch3n/vitest-ext-basic-example):

1. Go to `add.test.ts`
2. Add the following code (e.g. right above or below the tests):
    ```ts
    function testSubject<T>(value: T): [T] {
      return [value];
    }
    type StringTestSubject = typeof testSubject<string>
    ```
3. The green checkmarks/play buttons next to `describe(…)` and `it(…)` are still displayed, but the vitest extension output shows `[Error (timestamp)] parse error`
4. Try adding another test, e.g.:
    ```ts
    it("2 + 2", () => {
      expect(2 + 2).toBe(4);
    });
    ```
5. The test is not picked up, there is no green play button next to it
6. Close VSCode and re-open it
7. No tests in the `add.test.ts` file are picked up at all and "parse error" is logged in the extension output.

This only happens when adding the type parameter to the function. Simply doing `typeof testSubject` works fine, but doesn't yield the type I'm interested in. This syntax has been valid TypeScript since at least v4.7.4.

**Expected behavior**
Using a statement like `typeof testSubject<string>` should not break test detection.

**Environment**
<details>
<summary>info.txt</summary>

```json
{
  "System": {
    "OS": "Linux 5.10 Manjaro Linux"
  },
  "Binaries": {
    "Node": {
      "version": "18.12.0",
      "path": "~/.n/bin/node"
    },
    "Yarn": {
      "version": "1.22.19",
      "path": "/usr/bin/yarn"
    },
    "npm": {
      "version": "8.19.2",
      "path": "~/.n/bin/npm"
    }
  },
  "IDEs": {
    "VSCode": {
      "version": "1.82.2",
      "path": "/var/lib/snapd/snap/bin/code"
    }
  },
  "npmPackages": {
    "vite": {
      "installed": "2.9.16",
      "wanted": "^2.9.9"
    },
    "vitest": {
      "installed": "0.12.10",
      "wanted": "^0.12.6"
    }
  }
}
```

</details>

**Comment by MilanKovacic** [comment]
@ffMathy Could this be a similar issue as with the "using" keyword?

**Comment by ffMathy** [comment]
Possibly. Not sure.

**Comment by sheremet-va** [comment]
This should be fixed in pre-release 0.5.0 and higher. Note that the extension now requires Vitest 1.4.0 or higher.
